src/ray/raylet/worker_pool.cc

```diff
@@ -285,15 +345,32 @@ std::string WorkerPool::WarningAboutSize() {
   int64_t multiple = num_workers_started_or_registered / multiple_for_warning_;
   std::stringstream warning_message;
   if (multiple >= 3 && multiple > last_warning_multiple_) {
+    // Push an error message to the user if the worker pool tells us that it is
+    // getting too big.
     last_warning_multiple_ = multiple;
     warning_message << "WARNING: " << num_workers_started_or_registered
                     << " workers have been started. This could be a result of using "
                     << "a large number of actors, or it could be a consequence of "
                     << "using nested tasks "
                     << "(see https://github.com/ray-project/ray/issues/3644) for "
                     << "some a discussion of workarounds.";
+
+    if (gcs_client_ != nullptr) {
+      RAY_CHECK_OK(gcs_client_->error_table().PushErrorToDriver(
+          DriverID::Nil(), "worker_pool_large", warning_message.str(),
+          current_time_ms()));
+    } else {
+      RAY_LOG(WARNING)
+          << "Failed to push error message to user since gcs client is null.";
```

Comment:
why would the client be null?

Reply:
MockWorkerPool may pass a null to the ctor of WorkerPool.

Reply:
Since WorkerPool is a base class, we are not able to assume that the users of the derived class(not only MockWorkerPool) don't pass the null to client, right?

Reply:
We can override this function in `MockWorkerPool`. 
Even if we don't override, we shouldn't log this warning in test either